File: backend-app/src/main/java/com/example/oauth/user/controllers/UserController.java
package com.example.oauth.user.controllers;

import com.example.oauth.user.models.UserEntity;
import com.example.oauth.user.services.UserEntityService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class UserController {

  @Autowired
  private UserEntityService userService;

  @RequestMapping(method = RequestMethod.POST, value = "/users/login")
  public boolean loginUser(@RequestBody UserEntity user) {
    return userService.getSingleUser(user);
  }

  @RequestMapping("/users/details/{userName}")
  public UserEntity getUserDetails(@PathVariable String userName) {
    UserEntity dbUser = userService.getUserDetails(userName);
    return new UserEntity(dbUser.getId(), dbUser.getEmail(), dbUser.getName());
  }
}

File: backend-app/src/main/kotlin/com/example/oauth/user/controllers/UserController.kt
package com.example.oauth.user.controllers

import com.example.oauth.user.models.UserEntity
import com.example.oauth.user.services.UserEntityService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
class UserController {

    @Autowired
    var userEntityService: UserEntityService? = null;

    @RequestMapping(method = [RequestMethod.POST], value = ["/users/login"])
    fun loginUser(@RequestBody user: UserEntity): Boolean? {
        return userEntityService?.getSingleUser(user);
    }

    @RequestMapping("/users/details/{userName}")
    fun getUserDetails(@PathVariable userName: String): UserEntity {
        val dbUser: UserEntity = userEntityService!!.getUserDetails(userName);
        return UserEntity(dbUser.id, dbUser.email, dbUser.name);
    }
}